lib_bgp_data/blacklist/blacklist_parser.py

```diff
@@ -49,62 +50,77 @@ def _run(self):
+        output_path = dict()
+        # For each source in the sources dict, GET url, write response to path, and save path
         # to dict.
-        for source in sources.keys():
+        for source in sources:
             downloaded_file = requests.get(sources[source])
-            # UCEPROTECT is inconsistent in that sometimes a python
-            # requested file will sometimes come as a txt, sometimes
-            # as a tarball that needs to be unzipped. We check this:
-            if downloaded_file.headers['Content-Type'] == 'application/octet-stream':
-                decompressed = gzip.decompress(downloaded_file.content)
-                # UCEPROTECT uses ISO-8859-1 for text, not utf-8
-                output_str[source] = decompressed.decode('ISO-8859-1')
-            else:
-                # We have a txt and save to dict directly
-                output_str[source] = downloaded_file.text
-        return output_str
+            # Now we'll check if we got a proper file, if not try 5 more times or until success
+            if not downloaded_file.ok or downloaded_file.content == '':
+                for i in range(0, 5):
+                     # Don't want to spam the page
+                     time.sleep(5)
+                     downloaded_file = requests.get(sources[source])
+                     if downloaded_file.ok and downloaded_file.content != '':
+                         break
+            if not downloaded_file.ok or downloaded_file.content == '':
+                print("Aborting: File from " + source + 
+                      " failed to download properly with status code" + 
+                      str(downloaded_file.status_code) + "and length" + 
+                      str(len(downloaded_file.content)))
+                sys.exit()
+            _path = f"{self.path}/" + source
+            with open(_path, 'w+') as f:
+                f.write(downloaded_file.text)
+                output_path[source] = _path
+        return output_path
 
-    def _parse_lists(self, outputs: dict):
+    def _parse_lists(self, sources: dict):
         """For each source in outputs, takes the string attached to
         the source and parses it for ASNs, then saves to dict with
         sources as keys with a list of ASNs
         """
         parsed = dict()
         # For each source in outputs, parse for ASNs and save ASNs as
         # list for each source in dict.
-        for output in outputs.keys():
+        for source in sources:
             # If not a mit csv, just regex to find ASNs.
-            if output != 'mit':
-                parsed[output] = re.findall(r'AS\d+', outputs[output])
+            if source != 'mit':
+                with open(sources[source], 'r') as f:
+                    output = f.read()
+                    asns = re.findall(r'AS\d+', output)
```

Comment:
you use a regex here, and then a replace after? Look into capture groups, you shouldn't need a replace if you use a regex